Marlin/src/module/stepper.cpp

```diff
@@ -2025,21 +2030,38 @@ uint32_t Stepper::block_phase_isr() {
 
         // Update laser - Cruising
         #if ENABLED(LASER_POWER_INLINE_TRAPEZOID)
-          if (laser_trap.enabled) {
-            if (!laser_trap.cruise_set) {
-              laser_trap.cur_power = current_block->laser.power;
-              cutter.set_ocr_power(laser_trap.cur_power);
-              laser_trap.cruise_set = true;
+          if (cutter.cutter_mode == CUTTER_MODE_CONTINUOUS ) {
+            if (laser_trap.enabled) {
+              if (!laser_trap.cruise_set) {
+                laser_trap.cur_power = current_block->laser.power;
+                cutter.apply_power(laser_trap.cur_power);
+                laser_trap.cruise_set = true;
+              }
+              #if ENABLED(LASER_POWER_INLINE_TRAPEZOID_CONT)
+                laser_trap.till_update = LASER_POWER_INLINE_TRAPEZOID_CONT_PER;
+              #else
+                laser_trap.last_step_count = step_events_completed;
+              #endif
             }
-            #if ENABLED(LASER_POWER_INLINE_TRAPEZOID_CONT)
-              laser_trap.till_update = LASER_POWER_INLINE_TRAPEZOID_CONT_PER;
-            #else
-              laser_trap.last_step_count = step_events_completed;
-            #endif
           }
         #endif
       }
     }
+    #if ENABLED(LASER_FEATURE)
+      // Super fast method of dynamically adjusting the laser power ocr value based on the input feedrate in mm per minute
+      // TODO Setup Min/Max ocr ofsets to allow tuning and scaling of various lasers
+      // TODO Integrate accell/decell +-rate into the dynamic laser power calc
+      if (cutter.cutter_mode == CUTTER_MODE_DYNAMIC) {
+        if (planner.laser_inline.status.isPowered) {     // isPowered flag set on any parsed G1,G2,G3 or G5 move, cleared on any others.
+          if (last_laser_power != current_block->laser.power) {     // Prevent constant update without change
+            cutter.apply_power(current_block->laser.power);
+            last_laser_power = current_block->laser.power;
+          }
+        }
+      }
+  } else {
+      if (cutter.cutter_mode == CUTTER_MODE_DYNAMIC) cutter.apply_power(0);  //No movement in dynamic mode then turn Laser off
+    #endif
```

Comment:
This block is a mess and I cannot make sense of it to fix the indentation.

Reply:
Ok, I think I've sorted it out. It's helpful to keep blocks consistently following the cascade.